src/utils/operator.ts

```diff
@@ -40,13 +40,23 @@ function isOrderPartiallyFilled(order: RawOrder): boolean {
   }
 }
 
+interface RawOrderWithStatus extends RawOrder {
+  status: RawOrderStatusFromAPI
+}
+
+function hasRawOrderStatus(order: RawOrder): order is RawOrderWithStatus {
+  return (order as RawOrderWithStatus).status !== undefined
+}
+
 export function getOrderStatus(order: RawOrder): OrderStatus {
   if (isOrderFilled(order)) {
     return 'filled'
   } else if (order.invalidated) {
-    return 'canceled'
+    return 'cancelled'
   } else if (isOrderExpired(order)) {
     return 'expired'
+  } else if (hasRawOrderStatus(order) && order.status === RawOrderStatusFromAPI.presignaturePending) {
```

Comment:
I would prefer if you did a `isOrderPresigning` or similar to match the way the other status are checked.
And also have unit tests for the function.

Also, you shouldn't need `hasRawOrderStatus` as it should always be defined from the API. Or have you seen any cases where it's empty?
Regardless, comparing `undefined` === 'presignaturePending' will you give you the expected result.

Reply:
I have done the `isOrderPresigning` and created a test for this case.